packages/@aws-cdk/aws-appmesh/lib/gateway-route.ts

```diff
@@ -0,0 +1,390 @@
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnGatewayRoute } from './appmesh.generated';
+import { Protocol } from './shared-interfaces';
+import { IVirtualGateway, VirtualGateway } from './virtual-gateway';
+import { IVirtualService } from './virtual-service';
+
+/**
+ * Interface for which all GatewayRoute based classes MUST implement
+ */
+export interface IGatewayRoute extends cdk.IResource {
+  /**
+   * The name of the GatewayRoute
+   *
+   * @attribute
+   */
+  readonly gatewayRouteName: string,
+
+  /**
+   * The Amazon Resource Name (ARN) for the GatewayRoute
+   *
+   * @attribute
+   */
+  readonly gatewayRouteArn: string;
+
+  /**
+   * The VirtualGateway the GatewayRoute belongs to
+   *
+   * @attribute
+   */
+  readonly virtualGateway: IVirtualGateway;
+}
+
+/**
+ * Basic configuration properties for a GatewayRoute
+ */
+export interface GatewayRouteBaseProps {
+  /**
+   * The name of the GatewayRoute
+   *
+   * @default - an automatically generated name
+   */
+  readonly gatewayRouteName?: string;
+
+  /**
+   * What protocol the route uses
+   */
+  readonly routeSpec: GatewayRouteSpec;
+}
+
+/**
+ * The criterion for determining a request match for this GatewayRoute
+ */
+export interface HttpGatewayRouteMatch {
+  /**
+   * Specifies the path to match requests with.
+   * This parameter must always start with /, which by itself matches all requests to the virtual service name.
+   * You can also match for path-based routing of requests. For example, if your virtual service name is my-service.local
+   * and you want the route to match requests to my-service.local/metrics, your prefix should be /metrics.
+   *
+   */
+  readonly prefixPath: string;
+}
+
+/**
+ * The criterion for determining a request match for this GatewayRoute
+ */
+export interface GrpcGatewayRouteMatch {
```

Comment:
I wonder whether these `RouteMatch` interfaces bring a lot of value. While not have these properties directly in `HttpRouteSpecProps` / `GrpcRouteSpecProps`?

Reply:
So we will be building out different match types properties in the future. For example, you could theoretically match on request headers instead of the path. Would having them directly in the `SpecProps` lend itself to being extended easily in the future?

Reply:
OK, if you envision there will be a bunch of optional properties in the future in these `RouteMatch` interfaces, feel free to leave them as-is.